common/rect.h

```diff
@@ -243,41 +293,62 @@ struct Rect {
 		else if (right < r.left) right = r.left;
 	}
 
+   	/**
+	 * Reduce the dimensions of this rectangle by setting max width and max heigth.
+	 */
 	void clip(int16 maxw, int16 maxh) {
 		clip(Rect(0, 0, maxw, maxh));
 	}
 
+   	/**
+	 * @todo Doc required.
```

Comment:
Return true if the rectangle is empty (the width or the height is 0) or is invalid (negative width or height). Return false if this is a non-empty valid rectangle.